### DISCUSSION TITLE
Midi note input in Step Edit

### DISCUSSION THREAD

**Discussion by maks** [bangcorrupt/hacktribe#167] [Ideas]
I just got this idea watching a Syntakt review: It would be really nice to be able to enter a note for the currently selected step in StepEdit mode from an external midicontroller. I know Loopop documented [a hacky workaround for this using Step-Jump mode](https://www.youtube.com/watch?v=cEyxvnzoaiA), but I wonder if it would not be too hard to make this change in Hacktribe?

Ideally it would be nice to enter note(s) for a selected step via pads (per keyboard or chord mode style input) and selecting the current step in StepEdit mode instead with shift+pad, but I realise that's a way bigger ask to implement.

**Comment by bangcorrupt** [comment]
I plan to make Step Edit controllable via NRPN input at some point (same for all the pad modes).  Once that's done it shouldn't be too hard to link it to the onboard controls.

**Reply by maks** [reply]
That would be awesome! But with NRPN control, would note entry work with just standard midi note on-off commands?

**Reply by bangcorrupt** [reply]
Once it's working with NRPN it will be easy to switch it to note messages.  I'd like to set it up so it's easy to remap the MIDI control for everything, making it easier to interface with other devices that don't allow remapping.
